### ISSUE TITLE
[RLlib|New API|Inconsistency] LSTM Encoder lacks the output Linear, but stated in the docstring

### ISSUE THREAD

**Issue by brieyla1** [ray-project/ray#47625]
### **Description:**

There is an inconsistency between the docstring and the implementation of the `TorchLSTMEncoder` class in the RLlib new API. 

### **Current Behavior:**
The docstring of `TorchLSTMEncoder` states that the encoder has "One linear output layer". However, the current implementation does not include this linear output layer.

### **Expected Behavior:**
The implementation should match the docstring, either by:
1. Adding the linear output layer to the implementation, or
2. Removing the mention of the linear output layer from the docstring.

### **Files:**
`ray/rllib/core/models/tf/encoder.py`
`ray/rllib/core/models/torch/encoder.py`

### **Code Snippet:**

```python
class TorchLSTMEncoder(TorchModel, Encoder):
    """A recurrent LSTM encoder.

    This encoder has...
    - Zero or one tokenizers.
    - One or more LSTM layers.
    - One linear output layer.
    """
# ... rest of the implementation
@override(Model)
def _forward(self, inputs: dict, kwargs) -> dict:
    outputs = {}

    # ... existing code ...

    out, states_out = self.lstm(out, (states_in["h"], states_in["c"]))
    states_out = {"h": states_out[0], "c": states_out[1]}

    # !! Note: There's no application of a linear output layer here !!

    # Insert them into the output dict.
    outputs[ENCODER_OUT] = out
    outputs[Columns.STATE_OUT] = tree.map_structure(lambda s: s.transpose(0, 1), states_out)
    return outputs
```

### **Suggested Fix:**
Either:
1. Add a linear output layer in the `__init__` method and use it in the `_forward` method:
   ```python
   self.output_layer = nn.Linear(config.hidden_dim, config.output_dims[0])
   ```
   And in `_forward`:
   ```python
   out = self.output_layer(out)
   ```

2. Remove "- One linear output layer." from the docstring.

### **Additional Context:**
This inconsistency could lead to confusion for developers using or maintaining this encoder. Resolving it will improve code clarity and maintainability. This is the case in GRU and LSTM for Torch and TensorFlow Implementations. Is this what's supposed to happen?

**Action:**
I will be submitting a PR to fix this issue. The PR will be linked to this issue once created.

Labels: bug, documentation, inconsistency

### Versions / Dependencies

Latest Ray Github Version.

Other not relevant, but here it is:
Macos
Python 3.11 (using conda, and docker)

### Reproduction script

Go to the file located in 

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by brieyla1** [comment]
I've added the PR. Enjoy!

Please tell me if anything is missing. First time contributing publicly to the RLLIB repo

**Comment by simonsays1980** [comment]
@brieyla1 thanks for raising this issue. Great catch! I removed the parts from the docstrings that refer the linear output layer as this is indeed not present in our recurrent encoders. Because any encoder is paired in a model with a corresponding head at least a single linear layer follows always in our default implementations.
